docs/FullGuide.md

```diff
@@ -108,25 +103,88 @@ To be safe, check the state of the audio stream after you create it, as explaine
     }
 
 
-  3. You should verify the stream's configuration after opening it. If you specified
-  sample format, sample rate, or samples per frame they will not change.
-  However, sharing mode and buffer capacity might change (whether or not you set
-  them) depending on the capabilities of the stream's audio device and the
-  Android device on which it's running. As a matter of good defensive
-  programming, you should check the stream's configuration before using it.
-  There are functions to retrieve the stream setting that corresponds to each
-  builder setting:
+### Verifying Stream Configuration and additional properties
+
+You should verify the stream's configuration after opening it.
+
+The following properties are guaranteed. However, if these properties are unspecified,
+a default value will still be set, and should be queried by the appropriate
+accessor.
+
+* mStreamCallback (does not have an accessor)
+* mFramesPerCallback
+* mSampleRate
+* mChannelCount
+* mFormat
+* mDirection
+* mPerformanceMode
+
+The following properties may be changed by the underlying stream *even if
+explicitly set* and therefore should always be queried by the appropriate
+accessor.
+
+* mBufferCapacityInFrames
+* mSharingMode (ideally exclusive)
+
+The following properties are only set by the underlying stream. They cannot be
+set, but should be queried by the appropriate accessor.
+
+* mFramesPerBurst
+
+The following properties have unusual behavior
+
+* mDeviceId is respected by AAudio (API level >= 28), but not OpenSLES. It can
+be set regardless, but *will not* throw an error if an OpenSLES stream is used.
+
+* mAudioApi is only a property of the builder, however
+AudioStream::getAudioApi() can be used to query the underlying API which the
+stream uses. The property set in the builder is not guaranteed, and in
+general, the API should be chosen by Oboe to allow for best performance and
+stability considerations. Since Oboe is designed to be as uniform across both
+APIs as possible, this property should not generally be needed, however, it may
+be useful in the context of several known issues with OpenSLES (see bottom).
+* mBufferSizeInFrames can only be set on an already open stream (as opposed to a
+  builder), since it depends on run-time behavior. It can be set only up to the
+  BufferCapacity, and lacks an accessor.
+
+Since sharing mode and buffer capacity might change (whether or not you set
+them) depending on the capabilities of the stream's audio device and the
+Android device on which it's running, they must be queried. Additionally,
```

Comment:
Why must they be queried?

Reply:
Add Performance mode here

Reply:
Sorry, perhaps my comment wasn't clear. You say "..., they must be queried" but you don't explain _why_ they must be queried. This whole paragraph might be better worded as:

"Many of the stream's properties may vary (whether or not you set
them) depending on the capabilities of the audio device and the Android device on which it's running. If you need to know these values then you must query these using the accessor _after_ the stream has been opened."